sound/soc/intel/common/soc-acpi-intel-jsl-match.c

```diff
@@ -7,16 +7,22 @@
  */
 
 #include <sound/soc-acpi.h>
-#include <sound/soc-acpi-intel-match.h>
 
 struct snd_soc_acpi_mach snd_soc_acpi_intel_jsl_machines[] = {
 	{
-		.id = "DLGS7219",
+		.id = "MX98373",
 		.drv_name = "sof_da7219_max98373",
 		.machine_quirk = snd_soc_acpi_codec_list,
 		.sof_fw_filename = "sof-jsl.ri",
 		.sof_tplg_filename = "sof-jsl-da7219.tplg",
 	},
+	{
+		.id = "DLGS7219",
```

Comment:
it'd be a better idea to use the headset codec as a constant, and the amplifier name as a quirk in all cases. You are going to reach impossible combinations otherwise.

Reply:
The max98360a does not have programmable interface like i2c/gpio etc, so this spk amp exists anomalously without a name, that's why I put max98373 at top.

Reply:
ok, but why not follow the conventions used in the past. Just add DA7219+max98370 first, then DA7219 only, in this order. That will make it easier to deal e.g. with cases where we have a different amp.

Reply:
Sure, that should work.

Reply:
add a comment that the order matters please, so that we don't do silly edits later on

Reply:
Ack, will update.